babel/messages/pofile.py

```diff
@@ -80,6 +80,47 @@ def denormalize(string: str) -> str:
         return unescape(string)
 
 
+def _extract_locations(line: str) -> list[str]:
+    """Extract locations from location comments.
+
+    Locations are extracted while properly handling First Strong
+    Isolate (U+2068) and Pop Directional Isolate (U+2069), used by
+    gettext to enclose filenames with spaces and tabs in their names.
+    """
+    locations = []
```

Comment:
I think it might be a good idea to have a fast path here. Most projects will likely not have the isolate characters in `line`...

```suggestion
    if not ("\u2068" in line or "\u2069" in line):
        return line.lstrip().split()
    locations = []
```

or similar (the `line.lstrip().split()` being what we used to do).

---

Testing against [this large-ish translation file](https://raw.githubusercontent.com/bit-team/backintime/dev/common/po/fi.po):

### readpobench.py

```python
import io

from babel.messages.pofile import read_po

po_data = open("fi.po", "rb").read()


def fun():
    assert read_po(io.BytesIO(po_data))


for x in range(500):
    fun()
```

```
$ hyperfine 'git checkout master && python readpobench.py' 'git checkout enclose-whitespaces && python readpobench.py' 'git checkout enclose-whitespaces-fast-path && python readpobench.py'
Benchmark 1: git checkout master && python readpobench.py
  Time (mean ± σ):      1.538 s ±  0.005 s    [User: 1.493 s, System: 0.042 s]
  Range (min … max):    1.529 s …  1.547 s    10 runs

Benchmark 2: git checkout enclose-whitespaces && python readpobench.py
  Time (mean ± σ):      1.744 s ±  0.022 s    [User: 1.690 s, System: 0.047 s]
  Range (min … max):    1.710 s …  1.784 s    10 runs

Benchmark 3: git checkout enclose-whitespaces-fast-path && python readpobench.py
  Time (mean ± σ):      1.555 s ±  0.017 s    [User: 1.508 s, System: 0.042 s]
  Range (min … max):    1.537 s …  1.589 s    10 runs

Summary
  git checkout master && python readpobench.py ran
    1.01 ± 0.01 times faster than git checkout enclose-whitespaces-fast-path && python readpobench.py
    1.13 ± 0.02 times faster than git checkout enclose-whitespaces && python readpobench.py
```

Reply:
Sounds good to me. I just pushed a commit with slightly modified version of that.